src/monitoring/index.ts

```diff
@@ -0,0 +1,295 @@
+                type: IsomerHostedDomainType.KEYCDN,
+              } as IsomerHostedDomain)
+          )
+
+        return okAsync(domains)
+      })
+  }
+
+  getAmplifyDeployments() {
+    return this.launchesService.getAllDomains().map((domains) =>
+      domains.map(
+        (domain) =>
+          ({
+            domain,
+            type: IsomerHostedDomainType.AMPLIFY,
+          } as IsomerHostedDomain)
+      )
+    )
+  }
+
+  /**
+   * While most of our redirections are in our DB, we do have ad-hoc redirections.
+   * @returns List of redirection domains that are listed in the isomer-redirection repository
+   */
+  getRedirectionDomains() {
+    const SYSTEM_GITHUB_TOKEN = config.get("github.systemToken")
+    const OctokitRetry = Octokit.plugin()
+    const octokitWithRetry = new OctokitRetry({
+      auth: SYSTEM_GITHUB_TOKEN,
+      request: { retries: 5 },
+    })
+
+    return ResultAsync.fromPromise(
+      octokitWithRetry.request(
+        "GET /repos/opengovsg/isomer-redirection/contents/src/certbot-websites.csv"
+      ),
+      (error) =>
+        new MonitoringError(`Failed to fetch redirection domains: ${error}`)
+    )
+      .andThen((response) => {
+        const content = Buffer.from(response.data.content, "base64").toString(
+          "utf-8"
+        )
+        return ResultAsync.fromPromise(
+          new Promise<RedirectionDomain[]>((resolve, reject) => {
+            Papa.parse(content, {
+              header: true,
+              complete(results) {
+                // validate the csv
+                if (!results.data) {
+                  reject(new MonitoringError("Failed to parse csv"))
+                }
+                resolve(results.data as RedirectionDomain[])
+              },
+              error(error: unknown) {
+                reject(error)
+              },
+            })
+          }),
+          (error) => new MonitoringError(`Failed to parse csv: ${error}`)
+        )
+      })
+      .map((redirectionDomains) =>
+        redirectionDomains
+          .map((domain) => domain.source)
+          .map(
+            (domain) =>
+              ({
+                domain,
+                type: IsomerHostedDomainType.REDIRECTION,
+              } as IsomerHostedDomain)
+          )
+      )
+  }
+
+  /**
+   * This is in charge of fetching all the domains that are are under Isomer, inclusive
+   * of any subdomains and redirects.
+   */
+  getAllDomains() {
+    this.monitoringServiceLogger.info("Fetching all domains")
+    return ResultAsync.combine([
+      this.getAmplifyDeployments().mapErr(
+        (error) => new MonitoringError(error.message)
+      ),
+      this.getRedirectionDomains(),
+      this.getKeyCdnDomains(),
+    ]).andThen(([amplifyDeployments, redirectionDomains, keyCdnDomains]) => {
+      this.monitoringServiceLogger.info("Fetched all domains")
+      return okAsync(
+        [...amplifyDeployments, ...redirectionDomains, ...keyCdnDomains].sort(
```

Comment:
i think this might be a miscommunication - since our comparison is always based on the stripped length (ie, `www.isomer.gov.sg` is taken as equivalent to `isomer.gov.sg`), why not just use `sortBy(array, val => val.startsWith("www.") ? val.slice(0, 4) : val`.

in this case, our comparator compares via the stripped value so both `www.isomer.gov.sg` will be treated identically to `isomer.gov.sg`. lmk if this makes sense

Reply:
oo this is neater code, thanks